compatibility changes for dadbattle + philly

Lucky-56 opened the pull request:
fixes:

- 'Dad Battle' now has compatibility again *yay*

'Dad Battle' is now called 'Dad Battle' in game and uses 'dadbattle' as files
'Philly' is now called 'Philly Nice' in game and uses 'philly' as files (reverted #760 and instead fixed it better)

Lucky-56 committed:
compatibility changes for dadbattle + philly

'Dad Battle' is now called 'Dad Battle' in game and uses 'dadbattle' as files
'Philly' is now called 'Philly Nice' in game and uses 'philly' as files

puyoxyz commented on source/FreeplayState.hx:
```diff
@@ -202,13 +202,24 @@ class FreeplayState extends MusicBeatState
 		{
-			trace(StringTools.replace(songs[curSelected].songName," ", "-").toLowerCase());
+			// pre lowercasing the song name
+			var songLowercase = switch (songs[curSelected].songName)
```
This switch case is hella wack you're repeating code and changing like one argument

Do something like this instead

```haxe
var songLowercase = StringTools.replace(PlayState.SONG.song, " ", "-").toLowerCase();
switch (songLowercase) {
  case 'dad-battle': songLowercase = 'dadbattle';
  case 'philly-nice': songLowercase = 'philly';
}
```

puyoxyz commented on source/Paths.hx:
```diff
@@ -97,14 +97,30 @@ class Paths
 	{
-		song = StringTools.replace(song," ", "-");
-		return 'songs:assets/songs/${song.toLowerCase()}/Voices.$SOUND_EXT';
+		var songLowercase = switch (song)
```
See previous

puyoxyz commented on source/Paths.hx:
```diff
@@ -97,14 +97,30 @@ class Paths
 	{
-		song = StringTools.replace(song," ", "-");
-		return 'songs:assets/songs/${song.toLowerCase()}/Inst.$SOUND_EXT';
+		var songLowercase = switch (song)
```
see previous

puyoxyz commented on source/PlayState.hx:
```diff
@@ -236,14 +236,25 @@ class PlayState extends MusicBeatState
 		repPresses = 0;
 		repReleases = 0;
 
+		// pre lowercasing the song name (create)
```
see previous

puyoxyz commented on source/PlayState.hx:
```diff
@@ -1071,11 +1082,21 @@ class PlayState extends MusicBeatState
 		senpaiEvil.screenCenter();
 
-		if (SONG.song.toLowerCase() == 'roses' || SONG.song.toLowerCase() == 'thorns')
+		// pre lowercasing the song name (schoolIntro)
```
see previous

puyoxyz commented on source/PlayState.hx:
```diff
@@ -1368,9 +1389,20 @@ class PlayState extends MusicBeatState
 		var playerCounter:Int = 0;
 
+		// pre lowercasing the song name (generateSong)
+		var songLowercase = switch (PlayState.SONG.song)
```
see previous

puyoxyz commented on source/PlayState.hx:
```diff
@@ -2498,9 +2530,29 @@ class PlayState extends MusicBeatState
 
 					trace('LOADING NEXT SONG');
-					trace(PlayState.storyPlaylist[0].toLowerCase() + difficulty);
+					// pre lowercasing the next story song name
```
see previous

puyoxyz commented on source/PlayState.hx:
```diff
@@ -2498,9 +2530,29 @@ class PlayState extends MusicBeatState
 
-					if (SONG.song.toLowerCase() == 'eggnog')
+					// pre lowercasing the song name (endSong)
+					var songLowercase = switch (PlayState.SONG.song)
```
see previous

puyoxyz commented on source/Song.hx:
```diff
@@ -46,9 +46,23 @@ class Song
-		var rawJson = Assets.getText(Paths.json(StringTools.replace(folder," ", "-").toLowerCase() + '/' + StringTools.replace(jsonInput," ", "-").toLowerCase())).trim();
+		trace(jsonInput);
+		// pre lowercasing the folder name
+		var folderLowercase = switch (folder)
```
see previous

Lucky-56 committed:
cleaned up code

"This switch case is hella wack you're repeating code and changing like one argument"

Lucky-56 committed:
copy-pasted too much

puyoxyz reviewed (approved):
Looks good; @Lucky-56 did you test it with the new changes?

Lucky-56 committed:
fixed Freeplay

Lucky-56 committed:
forgot to call the song name

Lucky-56 committed:
fixed everything

highscore would be compatible if the package wasn't changed to KadeDev's own

Lucky-56 committed:
it's wierd, sometime it works sometimes not

Lucky-56 committed:
f... me

Lucky-56 committed:
will it stop crashing now?

puyoxyz commented on source/FreeplayState.hx:
```diff
@@ -202,13 +202,27 @@ class FreeplayState extends MusicBeatState
+			switch (songHighscore) {
+				case 'Dad-Battle': songHighscore = 'Dadbattle';
+				case 'Philly-Nice': songHighscore = 'Philly';
+			}
```
there *has to* be a less janky way to do this but this whole codebase is wack so whatever

puyoxyz reviewed (approved):
Yep

Lucky-56 committed:
Revert "fixed Tutorial having zooms on story mode"

This reverts commit a28ed62ee0b45a7b89d18c6f92ad76176458b703.

Lucky-56 committed:
fixed Tutorial having zooms on story mode

isakube reviewed (approved):
imagine making a commit and then immediately reverting it LOLL

RagdollClash reviewed (changes requested):
Satin Panties and Winter Horrorland also need compatibility changes.

RagdollClash reviewed (approved):
nvm, disregard what i said
